src/AspNetCore/Src/aspnetcoreconfig.cxx

```diff
@@ -8,16 +8,21 @@ ASPNETCORE_CONFIG::~ASPNETCORE_CONFIG()
     //
     // the destructor will be called once IIS decides to recycle the module context (i.e., application)
     //
+    // shutting down core application first
+    if (ASPNETCORE_APPLICATION::GetInstance() != NULL) {
```

Comment:
I'm not sure about if this is where dotnet.exe is shutdown, but it is triggered by the app pool being closed on IIS.